### PR TITLE
Add bytecode util to write primitive values with implicit casts

### PR DESCRIPTION
**Pull request by pranjalssh**
Add bytecode util to write primitive values in blockbuilder, along with implicit casts. For e.g. Allows us to call writeDouble(blockBuilder, long) directly, much like `TypeUtils.writeNativeValue()`

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-bytecode/src/main/java/com/facebook/presto/bytecode/StaticTypeBytecodeExpression.java
Comment (highker) [inline presto-bytecode/src/main/java/com/facebook/presto/bytecode/StaticTypeBytecodeExpression.java:97]: Maybe good to add a javadoc to explain this might be used by plugins? (Because it is not used anywhere in prestodb).

**Review by highker — APPROVED** [review]
approve for merging convenience. @kaikalur could have the final review. Feel free to ping other committers to merge while I'm not available.

**Review by kaikalur — APPROVED** [review]

**Comment by pranjalssh** [comment]
@highker can we merge?

**Pull request merged**
